src/basilisp/core.lpy

```diff
@@ -6784,7 +6794,7 @@
                               ["recmap" (->> (mapcat identity map#)
                                              (apply assoc ~'_recmap))])]
             (->> fields#
-                 (map (fn [[k# v#]] [(name k#) v#]))
+                 (map (fn [[k# v#]] [(.lstrip (-mangle-py-field '~type-name (munge k#)) "_") v#]))
```

Comment:
`attrs` actually strips leading underscores for `__init__` definitions, so we have to remove them for this case.